addonscript/addonscript/xpi.py

```diff
@@ -1,20 +0,0 @@
-"""Resources that operate on an XPI file."""
```

Comment:
I don't love that we're deleting and adding again, rather than moving.
Does it look like a new file to you, in your repo, if you diff from your head rev against `origin/master`? If it looks like it's been moved, maybe we can rebase-and-land rather than merge or squash-and-merge, and maybe that will preserve history. If it looks like it's been deleted and recreated, is there some way we can preserve that history? Would it work if we moved the files without any changes, and land that as a separate PR?

Reply:
I _think_ this should be fine as long as we don't squash merge. The diff for just that one rev looks much saner:
```
diff --git a/addonscript/addonscript/test/__init__.py b/addonscript/addonscript/test/__init__.py
deleted file mode 100644
index 794fdd9..0000000
--- a/addonscript/addonscript/test/__init__.py
+++ /dev/null
@@ -1,35 +0,0 @@
-"""Common utilities for the test harness."""
-import tempfile
-
-# import os
-import pytest
-
-# def read_file(path):
-#     with open(path, 'r') as fh:
-#         return fh.read()
-
-
-# BASE_DIR = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
-# TEST_DATA_DIR = os.path.join(os.path.dirname(__file__), "data")
-# PUB_KEY_PATH = os.path.join(TEST_DATA_DIR, "id_rsa.pub")
-
-
-# def noop_sync(*args, **kwargs):
-#     pass
-
-
-# async def noop_async(*args, **kwargs):
-#     pass
-
-
-@pytest.yield_fixture(scope='function')
-def tmpdir():
-    with tempfile.TemporaryDirectory() as tmp:
-        yield tmp
-
-
-# def is_slice_in_list(s, l):
-#     # Credit to https://stackoverflow.com/a/20789412/#answer-20789669
-#     # With edits by Callek to be py3 and pep8 compat
-#     len_s = len(s)  # so we don't recompute length of s on every iteration
-#     return any(s == l[i:len_s + i] for i in range(len(l) - len_s + 1))
diff --git a/addonscript/setup.py b/addonscript/setup.py
index 5170e65..31852e4 100644
--- a/addonscript/setup.py
+++ b/addonscript/setup.py
@@ -8,18 +8,19 @@ with open(os.path.join(os.path.abspath(os.path.dirname(__file__)), "version.txt"
 
 setup(
     name="addonscript",
     version=version,
     description="Script to submit unsigned addons to AMO and get signed copies back",
     author="Mozilla Release Engineering",
     author_email="release+python@mozilla.com",
     url="https://github.com/mozilla-releng/addonscript",
-    packages=find_packages(),
+    packages=find_packages("src"),
     package_data={"addonscript": ["data/*"]},
+    package_dir={"": "src"},
     include_package_data=True,
     zip_safe=False,
     entry_points={
         "console_scripts": [
             "addonscript = addonscript.script:main",
         ],
     },
     python_requires=">=3.6",
diff --git a/addonscript/addonscript/__init__.py b/addonscript/src/addonscript/__init__.py
similarity index 100%
rename from addonscript/addonscript/__init__.py
rename to addonscript/src/addonscript/__init__.py
diff --git a/addonscript/addonscript/api.py b/addonscript/src/addonscript/api.py
similarity index 100%
rename from addonscript/addonscript/api.py
rename to addonscript/src/addonscript/api.py
diff --git a/addonscript/addonscript/data/addonscript_task_schema.json b/addonscript/src/addonscript/data/addonscript_task_schema.json
similarity index 100%
rename from addonscript/addonscript/data/addonscript_task_schema.json
rename to addonscript/src/addonscript/data/addonscript_task_schema.json
diff --git a/addonscript/addonscript/exceptions.py b/addonscript/src/addonscript/exceptions.py
similarity index 100%
rename from addonscript/addonscript/exceptions.py
rename to addonscript/src/addonscript/exceptions.py
diff --git a/addonscript/addonscript/script.py b/addonscript/src/addonscript/script.py
similarity index 100%
rename from addonscript/addonscript/script.py
rename to addonscript/src/addonscript/script.py
diff --git a/addonscript/addonscript/task.py b/addonscript/src/addonscript/task.py
similarity index 100%
rename from addonscript/addonscript/task.py
rename to addonscript/src/addonscript/task.py
diff --git a/addonscript/addonscript/utils.py b/addonscript/src/addonscript/utils.py
similarity index 100%
rename from addonscript/addonscript/utils.py
rename to addonscript/src/addonscript/utils.py
diff --git a/addonscript/addonscript/xpi.py b/addonscript/src/addonscript/xpi.py
similarity index 100%
rename from addonscript/addonscript/xpi.py
rename to addonscript/src/addonscript/xpi.py
diff --git a/addonscript/tests/__init__.py b/addonscript/tests/__init__.py
new file mode 100644
index 0000000..e69de29
diff --git a/addonscript/tests/conftest.py b/addonscript/tests/conftest.py
new file mode 100644
index 0000000..407b436
--- /dev/null
+++ b/addonscript/tests/conftest.py
@@ -0,0 +1,9 @@
+import tempfile
+
+import pytest
+
+
+@pytest.yield_fixture(scope='function')
+def tmpdir():
+    with tempfile.TemporaryDirectory() as tmp:
+        yield tmp
diff --git a/addonscript/addonscript/test/test_api.py b/addonscript/tests/test_api.py
similarity index 99%
rename from addonscript/addonscript/test/test_api.py
rename to addonscript/tests/test_api.py
index 5c92356..0e6844a 100644
--- a/addonscript/addonscript/test/test_api.py
+++ b/addonscript/tests/test_api.py
@@ -3,19 +3,16 @@ import os
 
 import aiohttp
 import pytest
 from aioresponses import aioresponses
 from scriptworker.context import Context
 
 import addonscript.api as api
 from addonscript.exceptions import AMOConflictError, FatalSignatureError, SignatureError
-from addonscript.test import tmpdir
-
-assert tmpdir  # silence flake8
 
 
 @pytest.fixture(scope='function')
 def context():
     context = Context()
     context.config = {
         'amo_instances': {
             'project:releng:addons.mozilla.org:server:dev': {
diff --git a/addonscript/addonscript/test/test_script.py b/addonscript/tests/test_script.py
similarity index 100%
rename from addonscript/addonscript/test/test_script.py
rename to addonscript/tests/test_script.py
diff --git a/addonscript/addonscript/test/test_task.py b/addonscript/tests/test_task.py
similarity index 98%
rename from addonscript/addonscript/test/test_task.py
rename to addonscript/tests/test_task.py
index c13b783..2c3eaa3 100644
--- a/addonscript/addonscript/test/test_task.py
+++ b/addonscript/tests/test_task.py
@@ -1,18 +1,15 @@
 import os
 
 import pytest
 from scriptworker.context import Context
 from scriptworker.exceptions import TaskVerificationError
 
 import addonscript.task as task
-from addonscript.test import tmpdir
-
-assert tmpdir  # silence flake8
 
 
 @pytest.fixture(scope='function')
 def context(tmpdir):
     context = Context()
     context.config = {
         'amo_instances': {
             'project:releng:addons.mozilla.org:server:dev': {
diff --git a/addonscript/addonscript/test/test_utils.py b/addonscript/tests/test_utils.py
similarity index 99%
rename from addonscript/addonscript/test/test_utils.py
rename to addonscript/tests/test_utils.py
index 01f04d1..55bf901 100644
--- a/addonscript/addonscript/test/test_utils.py
+++ b/addonscript/tests/test_utils.py
@@ -8,19 +8,16 @@ import aiohttp
 import pytest
 from aioresponses import aioresponses
 from freezegun import freeze_time
 from jose import jws
 from jose.constants import ALGORITHMS
 from scriptworker.context import Context
 
 import addonscript.utils as utils
-from addonscript.test import tmpdir
-
-assert tmpdir  # silence flake8
 
 # helper constants, fixtures, functions {{{1
 # EXAMPLE_CONFIG = os.path.join(BASE_DIR, 'config_example.json')
 
 
 @pytest.fixture(scope='function')
 def context():
     context = Context()
```

Reply:
Ok. The merge commit in the scriptworker repo made it look like files were deleted and recreated, so maybe let's use the third option.